tests/integration/devfile/cmd_devfile_delete_test.go

```diff
@@ -31,187 +38,250 @@ var _ = Describe("odo devfile delete command tests", func() {
-				resourceTypes = append(resourceTypes, "routes")
-			}
-
-			helper.CmdShouldPass("odo", "push", "--project", commonVar.Project)
-
-			helper.CmdShouldPass("odo", "delete", "--project", commonVar.Project, "-f")
-
-			for _, resourceType := range resourceTypes {
-				commonVar.CliRunner.WaitAndCheckForExistence(resourceType, commonVar.Project, 1)
-			}
+		JustAfterEach(func() {})
+		It("should delete the component", func() {
+			helper.CmdShouldPass("odo", "delete", "-f")
 		})
 
-		It("should delete the component created from the devfile and also the env and odo folders and the odo-index-file.json file with all flag", func() {
-			helper.CmdShouldPass("odo", "create", "nodejs", "--project", commonVar.Project, componentName)
-			helper.CopyExample(filepath.Join("source", "devfiles", "nodejs", "project"), commonVar.Context)
-			helper.CopyExampleDevFile(filepath.Join("source", "devfiles", "nodejs", "devfile.yaml"), filepath.Join(commonVar.Context, "devfile.yaml"))
-			helper.CmdShouldPass("odo", "push", "--project", commonVar.Project)
-
-			helper.CmdShouldPass("odo", "url", "create", "example", "--host", "1.2.3.4.nip.io", "--port", "3000", "--ingress")
-
-			if os.Getenv("KUBERNETES") != "true" {
-				helper.CmdShouldPass("odo", "url", "create", "example-1", "--port", "3000")
-			}
+		When("the component is pushed", func() {
+			JustBeforeEach(func() {
+				helper.CmdShouldPass("odo", "push", "--project", commonVar.Project)
+			})
+			JustAfterEach(func() {})
+			It("should delete the component, env, odo folders and odo-index-file.json with --all flag", func() {
+				helper.CmdShouldPass("odo", "delete", "-f", "--all")
 
-			helper.CmdShouldPass("odo", "delete", "--project", commonVar.Project, "-f", "--all")
+				files := helper.ListFilesInDir(commonVar.Context)
+				Expect(files).To(Not(ContainElement(".odo")))
+				Expect(files).To(Not(ContainElement("devfile.yaml")))
+			})
 
-			commonVar.CliRunner.WaitAndCheckForExistence("deployments", commonVar.Project, 1)
+			Context("deleting a component from other component directory", func() {
+				var secondComp = helper.RandString(6)
+				var firstComp, firstDir, secondDir string
+
+				JustBeforeEach(func() {
+					// for the sake of verbosity
+					firstComp = componentName
+					firstDir = commonVar.Context
+					secondDir = createNewContext()
+					helper.Chdir(secondDir)
+				})
+				JustAfterEach(func() {
+					helper.Chdir(commonVar.Context)
+				})
+				When("the second component is created", func() {
+					JustBeforeEach(func() {
+						helper.CmdShouldPass("odo", "create", "nodejs", secondComp, "--project", commonVar.Project)
+					})
+					JustAfterEach(func() {})
+					When("the second component is pushed", func() {
+						JustBeforeEach(func() {
+							helper.CmdShouldPass("odo", "push", "--project", commonVar.Project)
+						})
+						JustAfterEach(func() {})
+						Context("delete with --context flag", func() {
```

Comment:
Is this information/Context necessary?

Reply:
I am not sure whether I understand your question correctly, but if what you mean to ask is whether we need a separate ginkgo.Context for this, then I would say yes, because in order to test this command with `--context` flag correctly, we either need to create /move to a new directory or move out of the component's directory.

Reply:
In line 246, you are using this Context, because you are changing the current directory in JustBeforeEach, which is fine.

But here, there is no JustBeforeEach, so the context would not change.

Reply:
Oh right right, it doesn't make sense to have that Context. I'll remove it, thanks.